compiler/rustc_middle/src/middle/stability.rs

```diff
@@ -406,7 +424,10 @@ impl<'tcx> TyCtxt<'tcx> {
                     }
                 }
 
-                EvalResult::Deny { feature, reason, issue, is_soft }
+                let using_vec_allocator_api = "std::vec::Vec::A"
```

Comment:
Can a diagnostic item be used here instead of this string comparison?

Reply:
@davidtwco 
Thank you so much for your review!
I applied your two above suggestions, but I am not sure how to implement this suggestion.

When checking stability for 'allocator_api', the span position has already moved to `Vec`'s second parameter, meaning nothing to indicate `Vec`. I could confirm that `Vec` appears before iteration of the `eval_stability` function, but I do not think it is better to preserve it only for 'allocator_api' with `Vec`.

Could you please tell me if there is a way to detect `Vec`, such as getting the parent type of generics? (If given `i16` from `Vec<i8, i16>`, reconstruct `Vec`.)

Reply:
@davidtwco 
What about showing this help on all use of unstable ‘allocator_api’ without focusing only on `Vec`?

Reply:
You should be able to do this with a [diagnostic item](https://rustc-dev-guide.rust-lang.org/diagnostics/diagnostic-items.html) - try placing a diagnostic item attribute on relevant generic arguments that indicate use of `allocator_api` and then checking if `def_id` has that item. You should be able to extend this approach to other uses of `allocator_api` without much difficulty.

I'd recommend moving the logic for this suggestion into a function that you invoke, just to keep the `eval_stability` function tidy.

Reply:
@davidtwco 
I was able to implement your suggestion with the diagnostic item! I appreciate your kind advice.